lib/mongoose-store.js

```diff
@@ -1,94 +1,64 @@
 'use strict';
 
-var shortid = require('shortid');
 var _ = require('lodash');
-
-var extractListFields = function(records) {
-
-  if (!_.isArray(records)) {
-    return records;
-  }
-
-  var list = [];
-  records.forEach(function(rec) {
-    list.push(rec.toObject());
-  });
-
-  return list;
-};
+var Promise = require('bluebird');
 
 function Store(_datasetId, _model) {
   this.model = _model;
   this.datasetId = _datasetId;
 }
 
-// do not want to create a new record on creation of a new store
 Store.prototype.init = function(data) {
-  console.log('mongoose-store doing init--- ', this.datasetId + ' : ' + data);
-  //var record = new this.model(data);
-
-  return {};//record.save();
+  if (!_.isArray(data)) {
+    console.log("Initialization data is not array.");
+    return Promise.resolve();
+  }
+  var self = this;
+  var promises = [];
+  _.each(data, function(entry) {
+    var record = new self.model(entry);
+    promises.push(record.save());
+  });
+  return Promise.all(promises);
 };
 
 Store.prototype.isPersistent = true;
 
 Store.prototype.create = function(object) {
-  console.log('do create--- ', this.datasetId + ' :object: ' + object);
-  object.id = shortid.generate();
-  var record = new this.model(_.clone(object));
-  return record.save(); //.then(_.property('fields'));
+  var record = new this.model(object);
+  return record.save();
 };
 
 Store.prototype.findById = function(id) {
-  console.log('do findById--- ', this.datasetId + ' :id: ' + id);
   // use 'list' since 'read' can only filter by guid === mongo's ObjectId
-  return this.model.findOne({id: id}).then(extractListFields);
+  return this.model.findOne({id: id}).lean();
```

Comment:
That's something we need to rethink as mongose related fields would introduce problems with sync (conflicts). For now we assume that we are only interested in plain objects, which is way faster than wrapping them into the models. We not going to use models in any way as they are often passed unchanged with little to no changes that needed to be passed back to db. I think that lean is the best option to use for the moment considering limitations and requirements.